prov/sm2/src/sm2_coordination.c

```diff
@@ -268,8 +268,13 @@ ssize_t sm2_file_open_or_create(struct sm2_mmap *map_shared)
 		}
 
 		common_fd = open(SM2_COORDINATION_FILE, O_RDWR);
-		if (common_fd > 0) {
+		if (common_fd >= 0) {
```

Comment:
This looks like a separate bugfix... plz put in different commit.

Reply:
3 commits now.